Leading dot in efs_facts.mountpoint

avaika opened the pull request:
##### SUMMARY
efs_facts module returns efs_facts.mount_point value with leading dot in the address.
It is impossible to pass the value to e.g. mount ansible module or use it other way without removing the dot in advance.
Didn't find any hint in the code on why the dot should be there.

##### ISSUE TYPE
 - Bugfix Pull Request (there was no ticket for it)

##### COMPONENT NAME
efs_facts
efs

##### ANSIBLE VERSION
```
ansible 2.4.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14+ (default, Dec  5 2017, 15:17:02) [GCC 7.2.0]
```

##### ADDITIONAL INFORMATION
Actual return information:
mount_point | .fs-xxxxxxxx.efs.us-west-2.amazonaws.com:/

Expected return information:
mount_point | fs-xxxxxxxx.efs.us-west-2.amazonaws.com:/

If the dot should be there, we need to add an explanation on why it is there and how one can live with it.

avaika committed:
remove leading dot in efs_facts mountpoint

avaika committed:
remove leading dot in efs_facts mountpoint

flowerysong commented:
This is a legacy of the original EFS design when it was released; the filesystem itself did not have a DNS name and you instead had to use a mount target DNS name (_availability-zone.file-system-id_.efs.aws-region.amazonaws.com) corresponding to your instance's AZ. At the time it was clear to anyone using the module why it returned a leading dot, since it always required modification before it could be used to mount the filesystem. Now that filesystem DNS names exist and are the default way to mount EFS volumes it's easy to miss the legacy DNS names in Amazon's documentation and less convenient to use the `mount_point` information.

It might be nicer to add the file system name as a new key instead of making a breaking change to the module's interface, but it's flagged as preview so I don't have a strong opinion. Note that whatever interface change is made, it should also be applied to the efs module.

avaika committed:
remove leading dot in efs.py

avaika commented:
@flowerysong , yes, you're right, I missed it in efs.py. Updated it as well.

Thank you for the background on why it is designed this way. The change indeed seems like a breaking one now. From the other side, it might feel a bit strange to introduce additional variable with almost the same value only for a compatibility purpose.

ryansb commented:
I think I agree with a non-breaking version of this change, where a new key (such as `filesystem_name` or similar) would mean anyone using the old way or that already has workarounds removing the dot would not be affected, and new users would just use `filesystem_name`. 

Also would need docs pointing people to use filesystem_name for new playbooks, but it would mean we could avoid having people modify their playbooks for the new version.

avaika committed:
introduce filesystem_address var in efs* modules and revert changes to mount_point

avaika commented:
Ok, it's a valid point. I reverted all the changes to mount_point, so it will return the value with leading dot like it does now.

+ I addedd filesystem_address variable without a leading dot and added some comments to the code. Hope it's better this way.

avaika commented:
Is there anything I can help with to get it merged?

willthames commented:
rebuild_merge

willthames commented:
Thanks for this contribution @avaika, sorry it took a while to get merged
